lib/gameState/HighScore.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * HighScore.h, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#pragma once
+
+VCMI_LIB_NAMESPACE_BEGIN
+
+class DLL_LINKAGE HighScoreParameter
+{
+public:
+	int difficulty;
+	int day;
+	int townAmount;
+	bool usedCheat;
+	bool hasGrail;
```

Comment:
Just in case - can you double-check that this flag works? There was report somewhere that it was broken (unconfirmed)

Reply:
Checked following:
- no grail
- grail in hero inventar
- grail in hero inventar + hero in town
- grail as building in town

For the three last you will get the correct 25 points. Seems correct. Cannot replicate the reported error.

Just as reference, report was:

> Now for my question: the last game I played was a random map generated with a Spider template. Free for all, me and 7 npc:s.
> First time the game ended I did not dig the grail and got some amount of points. Then I decided to try and find the grail from the map so I loaded an autosave, dug around and found it pretty fast and ended the game. The score, however, did not go up as I think it should have gone. Instead, I lost 3 points from the final score which is due to the couple of days of digging.